infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -222,6 +234,38 @@ private void initWriterAndBufferMgr() {
                 metadataManager.getReplicationMetadata(session).getLastLogEntryBatchProcessed(), this);
     }
 
+    private void specialInitForRoutingQModel() {
+        if (!session.getSubscriber().getModel().equals(LogReplication.ReplicationModel.ROUTING_QUEUES)) {
+            return;
+        }
+        insertQInRegistryTable();
+        insertTagForReplicatedQ();
+    }
+
+    private void insertQInRegistryTable() {
+        TableRegistry tableRegistry = runtime.getTableRegistry();
+        String replicatedQName = REPLICATED_QUEUE_NAME_PREFIX + session.getSourceClusterId();
+        if (!tableRegistry.getRegistryTable().containsKey(replicatedQName)) {
+            try {
+                tableRegistry.registerTable(CORFU_SYSTEM_NAMESPACE, replicatedQName, Queue.CorfuGuidMsg.class,
```

Comment:
Quick Question: registerTable works for queues as well?

I thought, we need to use openQueue like below to register the queue

            routingQueue =
                    corfuStore.openQueue(namespace, recvQueueName,
                            Queue.RoutingTableEntryMsg.class,
                            TableOptions.builder().schemaOptions(
                                            CorfuOptions.SchemaOptions.newBuilder()
                                                    .addStreamTag(LogReplicationUtils.REPLICATED_QUEUE_TAG)
                                                    .build())

Reply:
yes, `openQueue()` calls `openTable()` which inserts a record into registryTable.

Reply:
why do we invoke registerTable directly?  Cant openQueue be used?

Reply:
We did not want to open the queue...just wanted the registry table to have this queue so checkpointing happens. But this criteria could change with the discussion we had this morning.
But we still need to chalk all the details of that

Reply:
concluding on the open item:
We don't want LR to open the table, but just have an entry in the registry table so the stream gets checkpointed.